File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/how-to--attach-managed-code-extensions-to-documents_2.vb
        Dim documentPath As String = System.Environment.GetFolderPath( _
             Environment.SpecialFolder.Desktop) + "\WordDocument1.docx"
        Dim runtimeVersion As Integer = 0

        Try
            runtimeVersion = ServerDocument.GetCustomizationVersion(documentPath)

            ' Make sure that this document does not yet have any Visual Studio Tools 
            ' for Office customizations.
            If runtimeVersion = 0 Then
                Dim deployManifestPath As String = System.Environment.GetFolderPath( _
                    Environment.SpecialFolder.Desktop) & "\Publish\WordDocument1.vsto"
                Dim deploymentManifestUri As New Uri(deployManifestPath)
                ServerDocument.AddCustomization(documentPath, deploymentManifestUri)
                System.Windows.Forms.MessageBox.Show("The document was successfully customized.")
            Else
                System.Windows.Forms.MessageBox.Show("The document is already customized.")
            End If
        Catch ex As FileNotFoundException
            System.Windows.Forms.MessageBox.Show("The specified document does not exist.")
        Catch ex As DocumentNotCustomizedException
            System.Windows.Forms.MessageBox.Show("The document could not be customized." & _
                vbLf & ex.Message)
        End Try

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/how-to--attach-managed-code-extensions-to-documents_2.cs
            string documentPath = System.Environment.GetFolderPath(
                Environment.SpecialFolder.Desktop) + @"\WordDocument1.docx";
            int runtimeVersion = 0;

            try
            {
                runtimeVersion = ServerDocument.GetCustomizationVersion(documentPath);

                // Make sure that this document does not yet have any Visual Studio Tools 
                // for Office customizations.
                if (runtimeVersion == 0)
                {
                    string deployManifestPath = System.Environment.GetFolderPath(
                        Environment.SpecialFolder.Desktop) + @"\Publish\WordDocument1.vsto";

                    Uri deploymentManifestUri = new Uri(deployManifestPath);
                    ServerDocument.AddCustomization(documentPath, deploymentManifestUri);
                    System.Windows.Forms.MessageBox.Show("The document was successfully customized.");
                }
                else
                {
                    System.Windows.Forms.MessageBox.Show("The document is already customized.");
                }
            }
            catch (FileNotFoundException)
            {
                System.Windows.Forms.MessageBox.Show("The specified document does not exist.");
            }
            catch (DocumentNotCustomizedException ex)
            {
                System.Windows.Forms.MessageBox.Show("The document could not be customized.\n" +
                    ex.Message);
            }